services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py

```diff
@@ -269,6 +273,30 @@ async def get_entrypoint_container_name(
             log_httpx_http_error(url, "GET", traceback.format_exc())
             raise
 
+    async def restart_containers(self, dynamic_sidecar_endpoint: str) -> None:
+        """
+        runs docker-compose stop and docker-compose start in succession
+        resulting in a container restart without loosing state
+        """
+        url = get_url(
+            dynamic_sidecar_endpoint, f"/{self.API_VERSION}/containers:restart"
+        )
+        try:
+            async with httpx.AsyncClient(
+                timeout=self._restart_containers_timeout
+            ) as client:
+                response = await client.post(url=url)
```

Comment:
ready this [https://www.python-httpx.org/advanced/#more-efficient-usage-of-network-resources](https://www.python-httpx.org/advanced/#more-efficient-usage-of-network-resources)

I think it is not a good idea to create a new client each time, as internally it has a pool. they say ```significant performance improvements``` if you do not do that.

Reply:
For some reason it made totally sense to see that this is the correct way. But now I agree with you both. Will refactor.